server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessHealth.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.awarenesshealth;
+
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.routing.allocation.AwarenessReplicaBalance;
+import org.opensearch.cluster.routing.allocation.decider.AwarenessAllocationDecider;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Objects;
+
+/**
+ * Cluster state Awareness health information
+ *
+ */
+public class ClusterAwarenessHealth implements Writeable {
+
+    private final ClusterAwarenessAttributesHealth clusterAwarenessAttributesHealth;
+
+    /**
+     * Creates cluster awareness health from cluster state.
+     *
+     * @param clusterState The current cluster state. Must not be null.
+     * @param clusterSettings the current cluster settings.
+     * @param awarenessAttributeName Name of awareness attribute for which we need to see the health
+     */
+    public ClusterAwarenessHealth(
+        ClusterState clusterState,
+        ClusterSettings clusterSettings,
+        String awarenessAttributeName,
+        int totalShards
```

Comment:
why do we need total shards as an additional input when cluster state already has it

Reply:
Make sense will pull from cluster metadata itself.